### ISSUE TITLE
[Docs]: aws_ec2_transit_gateway_attachments example usage is using the wrong parameter for aws_ec2_transit_gateway_attachment

### ISSUE THREAD

**Issue by chuajiesheng** [hashicorp/terraform-provider-aws#35099]
### Documentation Link

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ec2_transit_gateway_attachments

### Description

By filter example usage is displaying

```
data "aws_ec2_transit_gateway_attachment" "unit" {
  count = length(data.aws_ec2_transit_gateway_attachments.filtered.ids)
  id    = data.aws_ec2_transit_gateway_attachments.filtered.ids[count.index]
}
```

It is supposed to be

```
data "aws_ec2_transit_gateway_attachment" "unit" {
  count = length(data.aws_ec2_transit_gateway_attachments.filtered.ids)
  transit_gateway_attachment_id    = data.aws_ec2_transit_gateway_attachments.filtered.ids[count.index]
}
```

### Would you like to implement a fix?

Yes

**Comment by McGon-Fid** [comment]
I will work on this and open a PR shortly. Thanks!

**Comment by johnsonaj** [comment]
closed in #35196
